### DISCUSSION TITLE
No details for metadata inconsistency error

### DISCUSSION THREAD

**Discussion by elenakorot** [hasura/graphql-engine#9536] [General] [answered]
When migrating changes between environments, if there are any inconsistencies in metadata / underlying DB structure, Hasura cloud just shows a generic error message ("cannot continue due to inconsistent metadata"):
![image](https://user-images.githubusercontent.com/129543920/229195213-5eb23eb8-4fef-4507-bf52-3ec8d4951e29.png)

It would be really helpful if the error message were more descriptive and included the details on the conflicts / issues (what needs to be fixed). 
Thank you!

**Comment by meetzaveri** [comment]
Hello @elenakorot 

Apologies for following late here. You can always find in-depth information for metadata inconsistency errors by visiting settings -> metadata -> metadata status. It will show you list of metadata with more descriptive information. URL - `/console/settings/metadata-status`.

**Comment by elenakorot** [comment]
Hi there,

Recently you had brought back the detailed error report on importing
metadata, but I no longer see it. Instead if there are any issues with the
metadata import, it just shows a generic message:

[image: image.png]

I am trying to import the metadata from one database to another. There is
no way to see what the actual inconsistencies are. Can you please bring
back the detailed error report, so that we can see what exactly needs to be
fixed.

Thank you!

Elena



On Thu, Apr 20, 2023 at 2:22 AM Meet Zaveri ***@***.***>
wrote:

> Hello @elenakorot <https://github.com/elenakorot>
>
> Apologies for following late here. You can always find in-depth
> information for metadata inconsistency errors by visiting settings ->
> metadata -> metadata status. It will show you list of metadata with more
> descriptive information. URL - /console/settings/metadata-status.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hasura/graphql-engine/discussions/9536#discussioncomment-5671307>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A64K54COXJS6F4332M2AUH3XCD55NANCNFSM6AAAAAAWO6WSJQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Elena Korotkina
Senior Software Engineer, MarketPush

(208) 398-3409‬  |  (972) 467-3909  |  www.marketpush.com

***@***.***  |  850 W Main St, Boise, ID 83702
<https://maps.google.com/?q=850%20W%20Main%20St,%20Boise,%20ID%2083702>
[image: App Banner Image] <https://www.marketpush.com/>

IMPORTANT: The contents of this email and any attachments are confidential.
They are intended for the named recipient(s) only. If you have received
this email by mistake, please notify the sender immediately and do not
disclose the contents to anyone or make copies thereof.

**Reply by meetzaveri** [reply]
Hey @vijayprasanna13 or @beaussan , can you provide some info here?

**Reply by meetzaveri** [reply]
Hey @elenakorot , can you tell us which version are you referring to ?

**Reply by vijayprasanna13** [reply]
> Recently you had brought back the detailed error report on importing
metadata, but I no longer see it. Instead if there are any issues with the
metadata import, it just shows a generic message:

@elenakorot are you referring to the UI where the inconsistencies are displayed in a table format with actions to delete/fix them. We have that for metadata that's already on your server and goes inconsistent during a restart. 
As for the detailed error report _on importing_ I don't think we do that. Like @meetzaveri said, if you could point us to the right version of Hasura I can verify if this is a regression.

**Comment by elenakorot** [comment]
Thank you!
Basically, I exported the metadata from one Hasura project and tried to
import it into another (the 2 projects have identical database structure),
and got an error message that the metadata import failed because of
inconsistent metadata but no other details (as far as that objects were
inconsistent so that I know what needs to be fixed).  You can see the error
message on the screenshot below.

I was able to get the detailed error message from the server response in
the browser dev tools:

[image: image.png]

We are using Hasura Cloud, v2.25 I believe. The issue is resolved since I
was able to get the details of the error and what needed to be fixed. But
if the error details can be provided in the UI / on the screen (without
having to look into the browser dev tools), that would be great!

Thank you again!





On Mon, May 15, 2023 at 10:24 PM Vijay Prasanna ***@***.***>
wrote:

> Recently you had brought back the detailed error report on importing
> metadata, but I no longer see it. Instead if there are any issues with the
> metadata import, it just shows a generic message:
>
> @elenakorot <https://github.com/elenakorot> are you referring to the UI
> where the inconsistencies are displayed in a table format with actions to
> delete/fix them. We have that for metadata that's already on your server
> and goes inconsistent during a restart.
> As for the detailed error report *on importing* I don't think we do that.
> Like @meetzaveri <https://github.com/meetzaveri> said, if you could point
> us to the right version of Hasura I can verify if this is a regression.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hasura/graphql-engine/discussions/9536#discussioncomment-5912182>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A64K54EZVY2USIIF2JRGISTXGMFRZANCNFSM6AAAAAAWO6WSJQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Elena Korotkina
Senior Software Engineer, MarketPush

(208) 398-3409‬  |  (972) 467-3909  |  www.marketpush.com

***@***.***  |  850 W Main St, Boise, ID 83702
<https://maps.google.com/?q=850%20W%20Main%20St,%20Boise,%20ID%2083702>
[image: App Banner Image] <https://www.marketpush.com/>

IMPORTANT: The contents of this email and any attachments are confidential.
They are intended for the named recipient(s) only. If you have received
this email by mistake, please notify the sender immediately and do not
disclose the contents to anyone or make copies thereof.

**Reply by vijayprasanna13** [reply, marked as answer]
Hey @elenakorot! This should be fixed on `v2.25.0-cloud.2` - 
![Screenshot 2023-05-18 at 3 32 51 PM](https://github.com/hasura/graphql-engine/assets/11921040/180726db-ac66-4cfa-a8df-eb4962e453ac)

**Comment by elenakorot** [comment]
Thank you!

On Thu, May 18, 2023 at 3:04 AM Vijay Prasanna ***@***.***>
wrote:

> Hey @elenakorot <https://github.com/elenakorot>! This should be fixed on
> v2.25.0-cloud.2 -
> [image: Screenshot 2023-05-18 at 3 32 51 PM]
> <https://user-images.githubusercontent.com/11921040/239204781-180726db-ac66-4cfa-a8df-eb4962e453ac.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hasura/graphql-engine/discussions/9536#discussioncomment-5937487>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A64K54ETE66GF63LQWZ2CGLXGXX4DANCNFSM6AAAAAAWO6WSJQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Elena Korotkina
Senior Software Engineer, MarketPush

(208) 398-3409‬  |  (972) 467-3909  |  www.marketpush.com

***@***.***  |  850 W Main St, Boise, ID 83702
<https://maps.google.com/?q=850%20W%20Main%20St,%20Boise,%20ID%2083702>
[image: App Banner Image] <https://www.marketpush.com/>

IMPORTANT: The contents of this email and any attachments are confidential.
They are intended for the named recipient(s) only. If you have received
this email by mistake, please notify the sender immediately and do not
disclose the contents to anyone or make copies thereof.

**Reply by meetzaveri** [reply]
Great, ~~I'll close this discussion for now~~. @elenakorot can you close this discussion for now ?
